File: Ejercicios Propuestos 2/Ejercicio 2/Ejercicio 2.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

# Solicitamos los datos
base = float(input("Ingrese la base (número real): "))
num = int(input("Ingrese el numerador del exponente: "))
den = int(input("Ingrese el denominador del exponente: "))

# Calculamos base^num
potencia = 1

for i in range(num):
    potencia *= base

# Calculamos la raíz den-ésima de potencia
raiz = potencia / den   # Valor inicial aproximado

# 50 iteraciones para mejorar la precisión
for i in range(50):
    raiz = ((den - 1) * raiz + (potencia / (raiz ** (den - 1)))) / den

resultado = raiz

# Mostramos el resultado
print(f"El resultado de {base} elevado a {num}/{den} es: {resultado}")

File: Ejercicios Propuestos 2/Ejercicio 2/ejercicio 2.cpp
#include <iostream>
#include <cmath> 

using namespace std;

int main() {
    // Definir base, num, den, potencia, resultado, raizz Como Real
    double base, num, den, potencia, resultado, raizz;
    // Definir i Como Entero
    int i;
    
    cout << "Ingrese la base (número real): ";
    cin >> base;
    
    cout << "Ingrese el numerador del exponente: ";
    cin >> num;
    
    cout << "Ingrese el denominador del exponente: ";
    cin >> den;
    
    // Calculamos base^num
    potencia = 1;
    for (i = 1; i <= num; i++) {
        potencia = potencia * base;
    }
    
    // Calculamos la raíz den-esima de potencia
    raizz = potencia / den;   // Valor inicial aproximado
    
    for (i = 1; i <= 50; i++) {   
        
        raizz = ( (den - 1) * raizz + (potencia / (pow(raizz, den - 1))) ) / den;
    }
    
    resultado = raizz;
    
    cout << "El resultado de " << base << " elevado a " << num << "/" << den << " es: " << resultado << endl;
    
    return 0;
}